networking/vm/kvm.go

```diff
@@ -417,339 +433,250 @@ func kvmTransformFlannelNetwork(net *activeNet) error {
-				return nil, errwrap.Wrap(errors.New("error in time of bridge setup"), err)
-			}
-			link, err := setupTapDevice(podID)
-			if err != nil {
-				return nil, errwrap.Wrap(errors.New("can not setup tap device"), err)
-			}
-			err = netlink.LinkSetMaster(link, br)
-			if err != nil {
-				rErr := tuntap.RemovePersistentIface(n.runtime.IfName, tuntap.Tap)
-				if rErr != nil {
-					stderr.PrintE("warning: could not cleanup tap interface", rErr)
-				}
-				return nil, errwrap.Wrap(errors.New("can not add tap interface to bridge"), err)
-			}
+		if err := addRoute(link, anet.Runtime.IP); err != nil {
+			return errwrap.Wrap(errors.New("cannot add on host direct route to pod"), err)
+		}
 
-			ifName := link.Attrs().Name
-			n.runtime.IfName = ifName
+	case "bridge":
+		config := BridgeNetConf{
+			BrName: defaultBrName,
+			MTU:    defaultMTU,
+		}
+		if err := json.Unmarshal(anet.ConfBytes, &config); err != nil {
+			return errwrap.Wrap(fmt.Errorf("error parsing %q result", anet.Conf.Name), err)
+		}
+		enableIpmasq = config.IPMasq
 
-			err = kvmSetupNetAddressing(&network, n, ifName)
-			if err != nil {
-				return nil, err
+		br, err := EnsureBridgeIsUp(config.BrName, config.MTU)
+		if err != nil {
+			return errwrap.Wrap(errors.New("error in time of bridge setup"), err)
+		}
+		link, err := setupTapDevice(n.Pod.UUID)
+		if err != nil {
+			return errwrap.Wrap(errors.New("can not setup tap device"), err)
+		}
+		err = netlink.LinkSetMaster(link, br)
+		if err != nil {
+			rErr := tuntap.RemovePersistentIface(anet.Runtime.IfName, tuntap.Tap)
+			if rErr != nil {
+				stderr.PrintE("warning: could not cleanup tap interface", rErr)
 			}
+			return errwrap.Wrap(errors.New("can not add tap interface to bridge"), err)
+		}
 
-			if n.conf.IsDefaultGateway {
-				n.runtime.IP4.Routes = append(
-					n.runtime.IP4.Routes,
-					cnitypes.Route{Dst: *defaultNet, GW: n.runtime.IP4.Gateway},
-				)
-				defaultGatewaySet = true
-				config.IsGw = true
-			}
+		ifName := link.Attrs().Name
+		anet.Runtime.IfName = ifName
 
-			if config.IsGw {
-				err = ensureHasAddr(
-					br,
-					&net.IPNet{
-						IP:   n.runtime.IP4.Gateway,
-						Mask: net.IPMask(n.runtime.Mask),
-					},
-				)
-
-				if err != nil {
-					return nil, errwrap.Wrap(fmt.Errorf("cannot add address to host bridge device %q", br.Name), err)
-				}
-			}
+		err = ExecNetIPAM(n, anet, ifName)
+		if err != nil {
+			return err
+		}
 
-		case "macvlan":
-			config := MacVTapNetConf{}
-			if err := json.Unmarshal(n.confBytes, &config); err != nil {
-				return nil, errwrap.Wrap(fmt.Errorf("error parsing %q result", n.conf.Name), err)
-			}
-			link, err := setupMacVTapDevice(podID, config, i)
-			if err != nil {
-				return nil, err
-			}
-			ifName := link.Attrs().Name
-			n.runtime.IfName = ifName
+		if config.IsGw {
+			anet.Runtime.IP4.Routes = append(
+				anet.Runtime.IP4.Routes,
+				cnitypes.Route{Dst: *defaultNet, GW: anet.Runtime.IP4.Gateway},
+			)
+			config.IsGw = true
```

Comment:
If the answer for above question is no - this is unnecessary line, and this if could be merged with next one.